plugins/modules/ndo_node_profile.py

```diff
@@ -0,0 +1,300 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2025, Samita Bhattacharjee (@samiib) <samitab@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: ndo_node_profile
+short_description: Manage Node Profiles on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage Node Profiles on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Samita Bhattacharjee (@samiib)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be a Fabric Resource Policy template.
+    type: str
+    required: true
+  name:
+    description:
+    - The name of the Node Profile.
+    type: str
+    aliases: [ node_profile ]
+  uuid:
+    description:
+    - The UUID of the Node Profile.
+    - This parameter is required when the O(name) needs to be updated.
+    type: str
+    aliases: [ node_profile_uuid ]
+  description:
+    description:
+    - The description of the Node Profile.
+    type: str
+  nodes:
+    description:
+    - The list of node IDs to associate with the Node Profile.
+    - This parameter is required when O(state=present).
```

Comment:
should this then also include the or here?
```suggestion
    - This parameter or O(node_setting_uuid) is required when O(state=present).
```

Reply:
`nodes` is always required when state is present. Only `node_setting` and `node_setting_uuid` are silently mutually exclusive as the UUID parameter takes precedence.

Reply:
you are right, I was looking at the wrong location thought this was the node_settings.